packages/model/src/-private/notify-changes.ts

```diff
@@ -16,26 +16,33 @@ export default function notifyChanges(
   record: Model,
   store: Store
 ) {
-  if (value === 'attributes') {
-    if (key) {
-      notifyAttribute(store, identifier, key, record);
-    } else {
-      record.eachAttribute((name) => {
-        notifyAttribute(store, identifier, name, record);
-      });
-    }
-  } else if (value === 'relationships') {
-    if (key) {
-      const meta = (record.constructor as typeof Model).relationshipsByName.get(key);
-      assert(`Expected to find a relationship for ${key} on ${identifier.type}`, meta);
-      notifyRelationship(identifier, key, record, meta);
-    } else {
-      record.eachRelationship((name, meta) => {
-        notifyRelationship(identifier, name, record, meta);
-      });
-    }
-  } else if (value === 'identity') {
-    record.notifyPropertyChange('id');
+  switch (value) {
+    case 'added':
```

Comment:
"added" is a scenario a record will never encounter unless it manages to get created in a prod build when no data exists for it through an illegal sync-relationship access.

so by subscribing to it here, we invalidate any attribute caches once we have data available, and only pay the cost in records which have done this vs needing to notify of attribute changes on first-load